### ISSUE TITLE
Build fails

### ISSUE THREAD

**Issue by frank-dittrich** [teeshop/rexgen#6]
With latest commit d29679a9db4672c0de91699f0213c4e5db7da37f I get:
```
(master)src $ ./clean.sh
(master)src $ ./build.sh 
entering /home/fd/git/rexgen/src/build
-- The C compiler identification is GNU 4.8.3
-- The CXX compiler identification is GNU 4.8.3
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
creating rexgen 1.2.1
-- COMPILING OPTIMIZED VERSION: RELEASE
CMake Error at librexgen/CMakeLists.txt:7 (include):
  include could not find load file:

    FindICU.cmake/FindICU.cmake


-- Found BISON: /usr/bin/bison (found suitable version "2.7", minimum required is "2.3") 
-- Found FLEX: /usr/bin/flex (found suitable version "2.5.37", minimum required is "2.5") 
-- Configuring incomplete, errors occurred!
See also "/home/fd/git/rexgen/src/build/CMakeFiles/CMakeOutput.log".
```

**Comment by frank-dittrich** [comment]
OK, I avoided this problem with a fresh clone, adding `--recursive`. 

```
git clone --recursive https://github.com/teeshop/rexgen.git
```

Now I get

```
(master)src $ ./clean.sh 
(master)src $ ./build.sh 
entering /home/fd/git/rexgen/src/build
-- The C compiler identification is GNU 4.8.3
-- The CXX compiler identification is GNU 4.8.3
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
creating rexgen 1.2.1
-- COMPILING OPTIMIZED VERSION: RELEASE
-- Found BISON: /usr/bin/bison (found suitable version "2.7", minimum required is "2.3") 
-- Found FLEX: /usr/bin/flex (found suitable version "2.5.37", minimum required is "2.5") 
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
ICU_INCLUDE_DIRS (ADVANCED)
   used as include directory in directory /home/fd/git/rexgen/src/librexgen
   used as include directory in directory /home/fd/git/rexgen/src/librexgen

-- Configuring incomplete, errors occurred!
See also "/home/fd/git/rexgen/src/build/CMakeFiles/CMakeOutput.log".
```

Looks like icu4c isn't an official fedora 20 package.

After downloading http://download.icu-project.org/files/icu4c/54.1/icu4c-54_1-RHEL6-x64.tgz, extracting and moving the files into the corresponding /usr/local/ subdirs, I get

```
(master)src $ ./clean.sh 
(master)src $ ./build.sh 
entering /home/fd/git/rexgen/src/build
-- The C compiler identification is GNU 4.8.3
-- The CXX compiler identification is GNU 4.8.3
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
creating rexgen 1.2.1
-- COMPILING OPTIMIZED VERSION: RELEASE
-- Found ICU: /usr/local/lib/libicuuc.so  
-- Found BISON: /usr/bin/bison (found suitable version "2.7", minimum required is "2.3") 
-- Found FLEX: /usr/bin/flex (found suitable version "2.5.37", minimum required is "2.5") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fd/git/rexgen/src/build
Scanning dependencies of target parser_target
[  3%] [BISON][regexparser] Building parser with bison 2.7
[  6%] [FLEX][regexscanner] Building scanner with flex 2.5.37
[  6%] Built target parser_target
Scanning dependencies of target librexgen
[ 10%] Building CXX object librexgen/CMakeFiles/librexgen.dir/parser.cpp.o
In file included from /home/fd/git/rexgen/src/librexgen/iterator/iterator.h:26:0,
                 from /home/fd/git/rexgen/src/librexgen/iterator/iteratorpermuter.h:26,
                 from /home/fd/git/rexgen/src/librexgen/regex/regex.h:24,
                 from /home/fd/git/rexgen/src/librexgen/regex/regexalternatives.h:25,
                 from parser/regex_parser.y:27:
/home/fd/git/rexgen/src/librexgen/string/simplestring.h:31:42: fatal error: librexgen/api/c/simplestring.h: No such file or directory
 #include <librexgen/api/c/simplestring.h>
                                          ^
compilation terminated.
make[2]: *** [librexgen/CMakeFiles/librexgen.dir/parser.cpp.o] Error 1
make[1]: *** [librexgen/CMakeFiles/librexgen.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by teeshop** [comment]
Hi,

I head the old headers still on my system, so that I didn't get the error.
I fixed this now.

Thank you, Jan

2015-02-24 11:01 GMT+01:00 frank-dittrich <notifications@github.com>:

**Comment by purnomoshinigami** [comment]
[ 83%] Building CXX object librexgen/CMakeFiles/librexgen.dir/c/simplestring.cpp.o
/blfs/rexgen/src/librexgen/c/simplestring.cpp: In function 'int c_simplestring_to_utf8_string(c_simplestring_ptr, char*, size_t)':
/blfs/rexgen/src/librexgen/c/simplestring.cpp:57:13: error: 'runtime_error' is not a member of 'std'
       throw std::runtime_error("mblen(): conversion error");
             ^~~
/blfs/rexgen/src/librexgen/c/simplestring.cpp:63:13: error: 'runtime_error' is not a member of 'std'
       throw std::runtime_error("wctomb(): conversion error");
             ^~~
make[2]: *** [librexgen/CMakeFiles/librexgen.dir/build.make:598: librexgen/CMakeFiles/librexgen.dir/c/simplestring.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:124: librexgen/CMakeFiles/librexgen.dir/all] Error 2
make: *** [Makefile:161: all] Error 2



add 
#include <stdexcept>
in 
rexgen/src/librexgen/c/simplestring.cpp

**Comment by teeshop** [comment]
thank you for reporting this. This is fixed as of 2.0.3 (i.e.: today ;-) )

**Comment by purnomoshinigami** [comment]
#include <stdexcept>
